backend/fms_core/template_importer/row_handlers/index_update/library.py

```diff
@@ -0,0 +1,68 @@
+            try:
+                derived_sample = DerivedSample.objects.get(samples__container__barcode=library["barcode"], samples__coordinate__name=library["coordinates"], library__index__name=index["old_index"], biosample__alias=library['alias'])
+                if derived_sample.library is not None:
+                    try:
+                        # get the new index id
+                        index = Index.objects.get(name=index["new_index"])
+                        # List all libraries that matches the targeted derived sample library
+                        # find root library derived sample
+                        root_library_derived_sample = None
+                        current_derived_sample = derived_sample
+                        while current_derived_sample and current_derived_sample.library is not None:
+                            root_library_derived_sample = current_derived_sample
+                            current_derived_sample = current_derived_sample.derived_from
+                        # descend the tree from root library
+                        def get_derived_samples(derived_sample: DerivedSample) -> Set[DerivedSample]:
+                            derived_samples = {derived_sample}
+                            for child_derived_sample in derived_sample.derived_to.all():
+                                derived_samples.update(get_derived_samples(child_derived_sample))
+                            return derived_samples
+                        
+                        derived_sample_lineage = get_derived_samples(root_library_derived_sample)
+
+                        # Apply the new index to the library identified
+                        for updated_derived_sample in derived_sample_lineage:
+                            _, self.errors["index_update"], self.warnings["index_update"] = update_library(derived_sample=updated_derived_sample, index=index)
+
+                        if len(derived_sample_lineage) > 1:
+                            self.warnings["multiple updates"] = f"A total of {len(derived_sample_lineage)} derived libraries that share lineage with the updated library were changed."
+                            
+                        # validate the current index change does not introduce index collision
+                        samples_impacted = Sample.objects.filter(derived_samples__in=derived_sample_lineage).distinct()
+                        for sample in samples_impacted:
+                            indices = Index.objects.filter(libraries__derived_sample__samples__id=sample.id)
+                            if len(indices) > 1:
+                                results, _, _ = validate_indices(indices=indices, threshold=0)
+                                if not results["is_valid"]:
+                                    self.warnings["index collision"].append(("Index {0} collide with an existing index in pooled library {1} with ID [{2}].", [index.name, sample.name, sample.id]))
+
```

Comment:
Instead of ID it could show barcode, coordinates, etc? We should introduce **markdown** support in our warnings at some point, maybe even have it support markdown table.